torch/optim/rmsprop.py

```diff
@@ -232,9 +240,14 @@ def _single_tensor_rmsprop(params: List[Tensor],
         if centered:
             grad_avg = grad_avgs[i]
             grad_avg.mul_(alpha).add_(grad, alpha=1 - alpha)
-            avg = square_avg.addcmul(grad_avg, grad_avg, value=-1).sqrt_().add_(eps)
+            avg = square_avg.addcmul(grad_avg, grad_avg, value=-1).sqrt_()
         else:
-            avg = square_avg.sqrt().add_(eps)
+            avg = square_avg.sqrt()
+
+        if differentiable:
+            avg = avg.clone().add_(eps)
```

Comment:
nit: just do an out of place add: `avg.add(eps)` ?

Reply:
yeah, right, i was in autopilot mode ...